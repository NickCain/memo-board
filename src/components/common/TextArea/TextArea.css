.textArea {
    min-height: 85px;
    background: transparent;
}

.textArea:focus{
    outline: 2px solid #d3d3d3;
}
