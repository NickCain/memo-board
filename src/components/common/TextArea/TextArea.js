import PropTypes from 'prop-types';
import React, { Component } from 'react';

import './TextArea.css';

class TextArea extends Component {
  render() {
    const { placeholder, onBlur, onKeyPress, autoFocus, text } = this.props;
    return (
      <textarea
        className="textArea"
        autoFocus={autoFocus}
        onBlur={onBlur}
        onKeyPress={onKeyPress}
        placeholder={placeholder}
        defaultValue={text}
      />
    );
  }
}

TextArea.propTypes = {
  text: PropTypes.string,
  placeholder: PropTypes.string,
  autoFocus: PropTypes.bool,
  onBlur: PropTypes.func,
  onKeyPress: PropTypes.func
};

export default TextArea;
