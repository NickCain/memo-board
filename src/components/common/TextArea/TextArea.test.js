import { mount, shallow } from 'enzyme';
import React from 'react';
import TextArea from './TextArea.js'

let wrapper;
let shallowWrapper;

describe('TextArea - component', () => {
  describe('TextArea renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <TextArea />
      );
      shallowWrapper = shallow(
        <TextArea />
      );
    });

    it('should render the TextArea component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });
  });
});
