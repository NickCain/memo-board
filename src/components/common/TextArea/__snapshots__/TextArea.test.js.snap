// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea - component TextArea renders should render the TextArea component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <textarea
    autoFocus={undefined}
    className="textArea"
    defaultValue={undefined}
    onBlur={undefined}
    onKeyPress={undefined}
    placeholder={undefined}
/>,
  "nodes": Array [
    <textarea
      autoFocus={undefined}
      className="textArea"
      defaultValue={undefined}
      onBlur={undefined}
      onKeyPress={undefined}
      placeholder={undefined}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TextArea />,
      "_debugID": 4,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TextArea {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <textarea
          autoFocus={undefined}
          className="textArea"
          defaultValue={undefined}
          onBlur={undefined}
          onKeyPress={undefined}
          placeholder={undefined}
/>,
        "_debugID": 5,
        "_renderedOutput": <textarea
          autoFocus={undefined}
          className="textArea"
          defaultValue={undefined}
          onBlur={undefined}
          onKeyPress={undefined}
          placeholder={undefined}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextArea />,
}
`;
